app/jobs/demo_mode/account_generation_job.rb

```diff
@@ -5,10 +5,18 @@ class AccountGenerationJob < DemoMode.base_job_name.constantize
     def perform(session, **options)
       session.with_lock do
         persona = session.persona
-        raise "Unknown persona: #{session.persona_name}" if persona.blank?
+        if persona.blank?
+          session.update!(failed_at: Time.current)
+          raise "Unknown persona: #{session.persona_name}"
```

Comment:
Since we're suppressing errors, shouldn't this be handled the same way we're handling other exceptions?

Reply:
I'm wondering if we can say something like:
```ruby
session.with_lock do
  persona = session.persona
  raise "Unknown persona: #{session.persona_name}" if persona.blank?

  signinable = persona.generate!(variant: session.variant, password: session.signinable_password, options: options)
  session.update!(signinable: signinable)
rescue StandardError => e
  session.update!(failed_at: Time.current)
  Rails.logger.error(e)
end
```